cmd/server/main.go

```diff
@@ -48,8 +48,7 @@ func (api *deploymentAPI) installerHandler(w http.ResponseWriter, r *http.Reques
 func (api *deploymentAPI) registerHandler(w http.ResponseWriter, r *http.Request) {
 	platform := r.FormValue("platform")
 	installerType := r.FormValue("installerType")
-	apiToken := r.FormValue("apiToken")
-
+	apiToken := r.Header.Get("Authorization")
```

Comment:
I suggest not changing the parameter for `/register` since it's only used for the tests to configure the mock, not used by the public Bosh release itself, so it's not a security concern.

If you still want to change it, then you'll need to modify the Ruby tests however, since they are the ones who call this `/register` endpoint:
* https://github.com/Dynatrace/bosh-oneagent-release/blob/decf3aa28d4c4273999b079084d5c16d45256f7b/spec/jobs/release_linux_spec.rb#L177
* https://github.com/Dynatrace/bosh-oneagent-release/blob/decf3aa28d4c4273999b079084d5c16d45256f7b/spec/jobs/release_windows_spec.rb#L177

Reply:
Added a check into registerHandler to make it compatible with current Bosh Ruby tests. What do you think about that?

Reply:
My position is that there is no need to make any changes on `registerHandler()` since it's only used for mocking purposes. That aside, Florian's comment remains, yes.